robot/Cumulus/tests/browser/batch_gift_entry/best_match_donation_matching_behaviour.robot

```diff
@@ -0,0 +1,126 @@
+    Verify Expected Values    nonns    npe01__OppPayment__c    ${pay_id}
+    ...    npe01__Payment_Amount__c=200.0
+    ...    npe01__Payment_Date__c=${date}
+    ...    npe01__Paid__c=True
+    ${opp_name}    Return Locator Value    check_field_spl    Opportunity
+    Click Link    ${opp_name}
+    ${opp_id} =           Get Current Record Id
+    Store Session Record      Opportunity  ${opp_id}
+    Confirm Value    Amount    $200.00    Y 
+    ${opp_date} =     Get Current Date    result_format=%-m/%-d/%Y
+    Confirm Value    Close Date    ${opp_date}    Y 
+    Confirm Value    Stage    Closed Won    Y
+    # Verify that the gift matched to existing opportunity and updated it to closed won status and payment is paid
+    Go To Record Home    &{opp_match}[Id]
+    Confirm Value    Amount    $100.00    Y 
+    ${opp_date} =     Get Current Date    result_format=%-m/%-d/%Y
+    Confirm Value    Close Date    ${opp_date}    Y 
+    Confirm Value    Stage    Closed Won    Y 
+    Select Tab    Related
+    Load Related List    GAU Allocations
+    Click Link    ${pay_no}
+    ${pay}    Get Current Record ID
+    Store Session Record      npe01__OppPayment__c  ${pay}
+    Verify Expected Values    nonns    npe01__OppPayment__c    ${pay}
+    ...    npe01__Payment_Amount__c=100.0
+    ...    npe01__Payment_Date__c=${date}
+    ...    npe01__Paid__c=True  
+    # Verify that the opportunity that does not match is still in prospecting stage
+    Go To Record Home    &{opp_dont_match}[Id]
+    Confirm Value    Amount    $50.00    Y 
+    ${opp_date} =     Get Current Date    result_format=%-m/%-d/%Y
+    Confirm Value    Close Date    ${opp_date}    Y 
+    Confirm Value    Stage    Prospecting    Y  
+
+***Keywords***
+Setup Test Data
+    ${ns} =  Get NPSP Namespace Prefix
+    Set Suite Variable    ${ns}
+    &{batch} =       API Create DataImportBatch    
+    ...    ${ns}Batch_Process_Size__c=50    
+    ...    ${ns}Batch_Description__c=Created via API    
+    ...    ${ns}Donation_Matching_Behavior__c=Best Match or Create    
+    ...    ${ns}Donation_Matching_Rule__c=${ns}donation_amount__c;${ns}donation_date__c    
+    ...    ${ns}RequireTotalMatch__c=false    
+    ...    ${ns}Run_Opportunity_Rollups_while_Processing__c=true   
+    ...    ${ns}GiftBatch__c=true    
+    ...    ${ns}Active_Fields__c=[{"label":"Donation Amount","name":"${ns}Donation_Amount__c","sObjectName":"Opportunity","defaultValue":null,"required":true,"hide":false,"sortOrder":0,"type":"number","options":null},{"label":"Donation Date","name":"${ns}Donation_Date__c","sObjectName":"Opportunity","defaultValue":null,"required":false,"hide":false,"sortOrder":1,"type":"date","options":null}] 
+    
+    Set Suite Variable    &{batch}
+    &{account} =     API Create Organization Account
+    Set Suite Variable    &{account}
+    ${date} =     Get Current Date    result_format=%Y-%m-%d
+    Set Suite Variable    ${date}
```

Comment:
I like how you set the suite variable right after the record is created, so that it deletes the session record anyway in case it fails in the middle of the tests case. :) I learnt this the hard way.